screen/wallets/addMultisigStep2.js

```diff
@@ -1,123 +1,59 @@
-import { HDSegwitBech32Wallet, MultisigCosigner, MultisigHDWallet } from '../../class';
+import { MultisigCosigner, MultisigHDWallet } from '../../class';
 import loc from '../../loc';
-import { SquareButton } from '../../components/SquareButton';
-import BottomModal from '../../components/BottomModal';
-import MultipleStepsListItem, {
-  MultipleStepsListItemButtohType,
-  MultipleStepsListItemDashType,
-} from '../../components/MultipleStepsListItem';
 import { BlueStorageContext } from '../../blue_modules/storage-context';
-import { encodeUR } from '../../blue_modules/ur';
+import { BlueURDecoder, encodeUR } from '../../blue_modules/ur';
 import QRCodeComponent from '../../components/QRCodeComponent';
 import alert from '../../components/Alert';
+import { Camera } from 'react-native-camera-kit';
+const createHash = require('create-hash');
 
 const prompt = require('../../helpers/prompt');
 const A = require('../../blue_modules/analytics');
-const fs = require('../../blue_modules/fs');
-const isDesktop = getSystemName() === 'Mac OS X';
 const staticCache = {};
+let decoder = false;
 
 const WalletsAddMultisigStep2 = () => {
-  const { wallets, addWallet, saveToDisk, isElectrumDisabled, isAdvancedModeEnabled, sleep } = useContext(BlueStorageContext);
+  const { wallets, addWallet, saveToDisk, isElectrumDisabled, sleep } = useContext(BlueStorageContext);
   const mainWallet = useMemo(() => wallets[0], [wallets]);
   const { colors } = useTheme();
-
   const navigation = useNavigation();
   const { m, n, format, walletLabel } = useRoute().params;
   const [cosigners, setCosigners] = useState([]); // array of cosigners user provided. if format [cosigner, fp, path]
   const [isLoading, setIsLoading] = useState(false);
-  const [isMnemonicsModalVisible, setIsMnemonicsModalVisible] = useState(false);
-  const [isProvideMnemonicsModalVisible, setIsProvideMnemonicsModalVisible] = useState(false);
-  const [isRenderCosignersXpubModalVisible, setIsRenderCosignersXpubModalVisible] = useState(false);
-  const [cosignerXpub, setCosignerXpub] = useState(''); // string used in exportCosigner()
   const [cosignerXpubURv2, setCosignerXpubURv2] = useState(''); // string displayed in renderCosignersXpubModal()
-  const [cosignerXpubFilename, setCosignerXpubFilename] = useState('bw-cosigner.json');
-  const [vaultKeyData] = useState({ keyIndex: 1, xpub: '', seed: '', isLoading: false }); // string rendered in modal
-  const [importText, setImportText] = useState('');
-  const [askPassphrase, setAskPassphrase] = useState(false);
-  const [isAdvancedModeEnabledRender, setIsAdvancedModeEnabledRender] = useState(false);
-  const openScannerButton = useRef();
-  const data = useRef(new Array(n));
-
-  useEffect(() => {
-    isAdvancedModeEnabled().then(setIsAdvancedModeEnabledRender);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  const scannedCache = {};
+  const quorum = useRef(new Array(n));
```

Comment:
Shouldn't the scannedCache be saved with a useState in order to be available via re-renders?

Reply:
Not really, this is actually a guard for not triggering a lot of renders, because scanning a QR code could potentially execute the callback multiple times for in the same pass

Reply:
Ok, I see 👍